conan/tools/cmake/cmakedeps2/target_configuration.py

```diff
@@ -482,6 +492,14 @@ def _template(self):
         {% endif %}
         {% endfor %}
 
+        # Definition of extra CMake variables from cmake_extra_variables
+
+        {% for key, value in extra_variables.items() %}
+        if (NOT CONAN_ORIGIN_EXTRA_VARIABLE_{{ key }}_IS_TOOLCHAIN)
```

Comment:
Isn't is possible instead of this protection to read the ``extra_variables =self._conanfile.conf.get("tools.cmake.cmaketoolchain:extra_variables")`` and avoid defining it completely?

This resolution in CMake seems a bit too delayed, the information is already in Conan, no need to pass it to CMake to decide.